modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessScriptEngine.java

```diff
@@ -101,9 +101,11 @@ public PainlessScriptEngine(Settings settings, Collection<ScriptContext<?>> cont
 
         for (ScriptContext<?> context : contexts) {
             if (context.instanceClazz.equals(SearchScript.class) || context.instanceClazz.equals(ExecutableScript.class)) {
-                contextsToCompilers.put(context, new Compiler(GenericElasticsearchScript.class, Definition.DEFINITION));
+                contextsToCompilers.put(context, new Compiler(GenericElasticsearchScript.class, new Definition(
```

Comment:
For now, let's move the construction of Definition out of the loop, so we still only create one, until we can figure out if we need an internal cache to ensure too many copies aren't loaded of common classes.

Reply:
Agreed.  Changed this to re-use the same Definition for all contexts for now.